modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/CurrencyConsensusHandler.scala

```diff
@@ -0,0 +1,41 @@
+package org.tessellation.currency.l0.snapshot
+
+import cats.effect.Async
+import cats.syntax.semigroupk._
+
+import org.tessellation.currency.l0.snapshot.schema.{CurrencyConsensusKind, CurrencyConsensusOutcome}
+import org.tessellation.currency.schema.currency.CurrencySnapshotContext
+import org.tessellation.node.shared.infrastructure.consensus.ConsensusRumorHandlers
+import org.tessellation.node.shared.infrastructure.gossip.RumorHandler
+import org.tessellation.node.shared.snapshot.currency._
+import org.tessellation.security.SecurityProvider
+
+import io.circe.Decoder
+
+object CurrencyConsensusHandler {
+  def make[F[_]: Async: SecurityProvider](
+    storage: CurrencyConsensusStorage[F],
+    manager: CurrencyConsensusManager[F],
+    fns: CurrencySnapshotConsensusFunctions[F]
+  )(implicit eventDecoder: Decoder[CurrencySnapshotEvent]): RumorHandler[F] = {
+    val all = new ConsensusRumorHandlers[
+      F,
+      CurrencySnapshotEvent,
+      CurrencySnapshotKey,
+      CurrencySnapshotArtifact,
+      CurrencySnapshotContext,
+      CurrencySnapshotStatus,
+      CurrencyConsensusOutcome,
+      CurrencyConsensusKind
+    ](storage, manager, fns)
+
+    all.eventHandler <+>
+      all.facilityHandler <+>
+      all.proposalHandler <+>
+      all.signatureHandler <+>
+      all.binarySignatureHandler <+>
```

Comment:
The difference between currency and global is using `binarySignatureHandler` for currency and not for global.